app/Api/Controllers/NotificationController.php

```diff
@@ -0,0 +1,74 @@
+<?php
+
+namespace App\Api\Controllers;
+
+use Dingo\Api\Http;
+use Dingo\Api\Routing\Helpers;
+use App\User;
+use App\Notification;
+use App\NotificationAttrib;
+use Illuminate\Http\Request;
+
+class NotificationController extends Controller
+{
+
+    use Helpers;
+
+    public function __construct() {
+    }
+
+    /**
+     * Display a listing of all notifications
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function index(Request $request, $type = null)
+    {
+        if ($type === 'archive')
+        {
+            //$notifications = Notification::read([$request->user()->user_id])->limit()->get();
+            $notifications = Notification::select('notifications.*','key')->leftJoin('notifications_attribs', 'notifications.notifications_id', '=', 'notifications_attribs.notifications_id')->where('user_id', $user_id)->where(['key'=>'read', 'value'=> 1]);
+        }
+        else {
+            //$notifications = Notification::unread()->limit()->get();
+            $notifications = Notification::select('notifications.*','key')->leftJoin('notifications_attribs', 'notifications.notifications_id', '=', 'notifications_attribs.notifications_id')->whereNull('user_id')->orWhere(['key'=>'sticky', 'value'=> 1])->get();
```

Comment:
So more like the commented out but rather than chaining the call just have it done so that unread() does it?